### ISSUE TITLE
sdk list suspends without output under zsh

### ISSUE THREAD

**Issue by cy6erGn0m** [sdkman/sdkman-cli#984]
**Bug report**
sdk list suspends without output under zsh (with and without Oh My Zsh) but works under bash

**To reproduce**
1. start zsh
2. sdk list java
3. Observe results:

```text
cy6ergn0m@white-angel:~> sdk list java
[1]  + 18028 suspended (tty output)  sdk list java
cy6ergn0m@white-angel:~> ps
  PID TTY          TIME CMD
17919 pts/9    00:00:00 zsh
18027 pts/9    00:00:00 less
18028 pts/9    00:00:00 zsh
18030 pts/9    00:00:00 ps
```

Please note the second zsh that appeared after running the command (had only one before)

**Workaround**
Executing `fg` after sdk get suspended does the trick and the list appears as it should.

**System info**

- openSUSE Leap 15.3
- zsh 5.6 (x86_64-suse-linux-gnu)
- SDKMAN 5.12.4

**Comment by marc0der** [comment]
Hi @cy6erGn0m, I'm also a Linux user with ZSH and oh-my-zsh, but never seem to have had this issue. I also went as far as trying to replicate this in Docker by running the following:

```
$ docker run -it --rm opensuse/leap:latest bash
# zypper refresh
# zypper install zip unzip curl zsh
# zsh
# curl -s https://get.sdkman.io | bash
# source /root/.sdkman/bin/sdkman-init.sh
# sdk ls java
```
This works perfectly for me without hanging at all. Could this be something local to your own environment?

**Comment by cy6erGn0m** [comment]
Just reproduced it under Windows with WSL2 at clean opensuse installation so I wonder why you didn't got it under docker. Will try in a real linux..

**Comment by marc0der** [comment]
Okay, please let me know any findings you make.

**Comment by cy6erGn0m** [comment]
First or all, in your example, you don't even have less therefore you don't get the error. You should install less as well. 

Finally, I found that the reason is that in both my Linux systems (Ubuntu and OpenSUSE) I have `$PAGER` env specified to "less" explicitly. Unsetting it also fixes the bug. However, `$PAGER` is quite gnuish and there is nothing weird about it so I suppose sdk-cli should work with it.

**Comment by cy6erGn0m** [comment]
So, the full repro is:

```
$ docker run -it --rm opensuse/leap:latest bash
# zypper refresh
# zypper install -y less zip unzip curl zsh
# zsh
# curl -s https://get.sdkman.io | bash
# source /root/.sdkman/bin/sdkman-init.sh
# export PAGER=less
# sdk ls java
```

**Comment by marc0der** [comment]
Hi @cy6erGn0m, this is very strange as I also have the `$PAGER` environment variable set to `less` on my system (Manjaro Linux) without any such issue. I also ran the same test as above on the latest Ubuntu in Docker (with the `PAGER` env var exported), and it worked perfectly.

So far, this only seems to be happening on OpenSuSE. Let's dig some more to see if we can figure this out.

**Comment by cy6erGn0m** [comment]
Actually, i also have it on Ubuntu

**Comment by marc0der** [comment]
> Actually, i also have it on Ubuntu

Super weird. Here is what I did to test it:

```
$ docker run -it --rm ubuntu:latest bash
# apt update && apt install -y zip unzip curl zsh less
# zsh
# export PAGER=less
# curl -s https://get.sdkman.io | bash
# source "/root/.sdkman/bin/sdkman-init.sh"
# sdk ls java
```

That seems to work perfectly :man_shrugging:

**Comment by helpermethod** [comment]
Hi @cy6erGn0m,

do you have any alias for `ls`, `list` or `java` defined?

Could you run the following line

    set - x; sdk list java; set +x

and share the output?

**Comment by cy6erGn0m** [comment]
Of course I do, at least for different kinds of `ls` but nothing special

```
⚙ cy6ergn0m@white-angel  ~/.sdkman  alias | grep -w ls
dir='ls -l'
gfg='git ls-files | grep'
gignored='git ls-files -v | grep "^[[:lower:]]"'
gwip='git add -A; git rm $(git ls-files --deleted) 2> /dev/null; git commit --no-verify --no-gpg-sign -m "--wip-- [skip ci]"'
l='ls -lah'
la='ls -lAh'
ll='ls -lh'
ls='ls --color=tty'
ls-l='ls -l'
lsa='ls -lah'
zls='zypper ls'
 ⚙ cy6ergn0m@white-angel  ~/.sdkman  alias | grep -w less
o=less
 ⚙ cy6ergn0m@white-angel  ~/.sdkman  alias | grep -w list
gcf='git config --list'
gdct='git describe --tags $(git rev-list --tags --max-count=1)'
gstl='git stash list'
hgun='hg resolve --list'
 ⚙ cy6ergn0m@white-angel  ~/.sdkman  
```

Also, see the debug output here:
https://gist.github.com/cy6erGn0m/c7c8ed14e55b04307d6bde6d259598cd

**Comment by cy6erGn0m** [comment]
Anyway, I don't get: if I run in docker, my envs shouldn't be applied inside the container so why I can repro and you can't with the same image?

**Comment by cy6erGn0m** [comment]
Inside the container I have much less envs:

```
root@845ba2369fa3:/ # env     
HOSTNAME=845ba2369fa3
GPG_TTY=/dev/pts/0
PWD=/
HOME=/root
TERM=xterm
SHLVL=2
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
_=/usr/bin/env
LOGNAME=root
OLDPWD=/
SDKMAN_VERSION=5.12.4
SDKMAN_CANDIDATES_API=https://api.sdkman.io/2
SDKMAN_DIR=/root/.sdkman
SDKMAN_PLATFORM=linuxx64
SDKMAN_CANDIDATES_DIR=/root/.sdkman/candidates
PAGER=less
root@845ba2369fa3:/ # alias
which-command=whence
root@845ba2369fa3:/ # cat /etc/os-release 
NAME="openSUSE Leap"
VERSION="15.3"
ID="opensuse-leap"

```

**Comment by cy6erGn0m** [comment]
Also tried different virtual terminals like konsole (my primary), rxvt and IDEA's terminal. No luck.

**Comment by cy6erGn0m** [comment]
I even examined less:

```
root@845ba2369fa3:/ # which less
/usr/bin/less
root@845ba2369fa3:/ # file /usr/bin/less
/usr/bin/less: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=5bdc1539ba5a4afebe8188d22674b79bc579ff6c, stripped
root@845ba2369fa3:/ # hexdump -C /usr/bin/less | head
00000000  7f 45 4c 46 02 01 01 00  00 00 00 00 00 00 00 00  |.ELF............|
00000010  03 00 3e 00 01 00 00 00  50 44 00 00 00 00 00 00  |..>.....PD......|
00000020  40 00 00 00 00 00 00 00  40 89 02 00 00 00 00 00  |@.......@.......|
00000030  00 00 00 00 40 00 38 00  09 00 40 00 1d 00 1c 00  |....@.8...@.....|
00000040  06 00 00 00 05 00 00 00  40 00 00 00 00 00 00 00  |........@.......|
00000050  40 00 00 00 00 00 00 00  40 00 00 00 00 00 00 00  |@.......@.......|
00000060  f8 01 00 00 00 00 00 00  f8 01 00 00 00 00 00 00  |................|
00000070  08 00 00 00 00 00 00 00  03 00 00 00 04 00 00 00  |................|
00000080  38 02 00 00 00 00 00 00  38 02 00 00 00 00 00 00  |8.......8.......|
00000090  38 02 00 00 00 00 00 00  1c 00 00 00 00 00 00 00  |8...............|
```

**Comment by cy6erGn0m** [comment]
Also tried to switch pager to `more` instead of `less`: it works the same: get suspended and now I see it is `more` utility hanging instead of `less`. So for sure `$PAGER` works and makes difference.

**Comment by cy6erGn0m** [comment]
stty options seems to be identical

**Comment by cy6erGn0m** [comment]
After digging more I discovered that the reason is somewhere around `eval "$PAGER"` that suddenly suspends with no visible reason. After quick searching I found that this could be a [zsh bug "Unexpected process suspension"](https://www.zsh.org/mla/workers/2020/msg00640.html). This seems to be fixed but... OpenSUSE Leap at the moment is one of the most conservative distros and still pinned on zsh 5.6.

Upgrading to zsh 5.8 via custom [repo](https://download.opensuse.org/repositories/shells/openSUSE_Leap_15.3/shells.repo) fixed the issue so I think we may close this ticket. Thank you for you support and patience!
